### PR TITLE
chore!: update android `minSdkVersion` to 23, kotlin to 1.8.22, gradle build tools to 7.4.2, and gradle to 7.5

### PR DESCRIPTION
**Pull request by bradrushworth**

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -4,13 +4,13 @@ version: 2.0.3
 homepage: https://github.com/zeshuaro/google_api_headers
 
 environment:
-  sdk: ">=2.12.0 <4.0.0"
+  sdk: '>=2.19.6 <4.0.0'
```
Comment (zeshuaro) [inline pubspec.yaml:7]: Why do we need to bump the lower bound of the supported sdk and specifically to 2.19.6?

File: pubspec.yaml [hunk 2]
```diff
@@ -4,13 +4,13 @@ version: 2.0.3
...
   flutter: ">=1.20.0"
 
 dependencies:
   flutter:
     sdk: flutter
-  package_info_plus: ^4.0.0
+  package_info_plus: ^4.2.0
```
Comment (zeshuaro) [inline pubspec.yaml:13]: If we do not need functionalities from 4.2.0, I would prefer keeping this as is instead of bumping the required version

Comment (bradrushworth) [reply pubspec.yaml:13]: My read of the changelog is there are bug fixes between those versions. Keeping legacy versions of kotlin and gradle can cause big problems for Android developers. This gradle/kotlin issue is why I made this PR in the first place. Android SDK version 34 is making big changes and legacy dependencies like this block people like me from upgrading.

I had to update a number of dependency projects like this one, so I don't exactly remember the specifics for this project though. Sorry.

**Pull request merged**